python/ray/data/dataset.py

```diff
@@ -776,6 +778,68 @@ def _map_batches_without_batch_size_validation(
         logical_plan = LogicalPlan(map_batches_op, self.context)
         return Dataset(plan, logical_plan)
 
+    @PublicAPI(api_group=EXPRESSION_API_GROUP, stability="alpha")
+    def with_columns(
+        self,
+        exprs: List[Expr],
+        batch_format: Optional[str] = "pyarrow",
```

Comment:
Let's not expose separate batch_format

 - We implement and support all expr infra
 - We should only support block-types
 - No additional conversions should occur